R/config.R

```diff
@@ -180,6 +248,15 @@ config_validate <- function(self, filename) {
 
   self$archive_version <- read_orderly_archive_version(".")
 
+  lockBinding(quote(destination), self)
```

Comment:
We already have one bit of locking in the initialize function, and I think this would go more naturally there.

Perhaps do this with a utility function that either locks all data members in a for loop or takes a vector of names and loops over that to lock them - we'll use this elsewhere